### PR TITLE
#924 Resign users absent from people.xml

### PR DESCRIPTION
**Pull request by carlosmiranda**
#924 Users who are absent from people.xml (deleted/kicked out from Zerocracy) are now deleted from projects.

Exception is users who have `PO` role. Some of our clients are logged in to Zerocracy but are not invited, so they are not counted in `people.xml`. Instead, they are added to projects manually as `PO`. We don't want to kick them out of their own projects, which might possibly become orphaned if they are resigned.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1685](https://github.com/zerocracy/farm/issues/1685) is now in scope, role is `REV`

File: src/main/groovy/com/zerocracy/stk/pm/staff/remove_stale_users.groovy [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import com.zerocracy.pm.staff.Roles
+import com.zerocracy.pmo.Catalog
+import com.zerocracy.pmo.People
+
+def exec(Project project, XML xml) {
+  new Assume(project, xml).notPmo()
+  new Assume(project, xml).type('Ping hourly')
+  Farm farm = binding.variables.farm
+  Catalog catalog = new Catalog(farm).bootstrap()
+  if (!catalog.sandbox().contains(project.pid())) {
```
Comment (g4s8) [inline src/main/groovy/com/zerocracy/stk/pm/staff/remove_stale_users.groovy]: @carlosmiranda why sandbox projects are skipped here? I think we should also resign roles from non-existing users.

Comment (carlosmiranda) [reply src/main/groovy/com/zerocracy/stk/pm/staff/remove_stale_users.groovy]: @g4s8 artifact from bad copypasta move. xD

**Comment by carlosmiranda** [comment]
@g4s8 updated, please check

**Comment by g4s8** [comment]
@rultor merge

**Comment by g4s8** [comment]
@rultor merge

**Comment by g4s8** [comment]
@carlosmiranda it seems that you have checkstyle issue:
```test/resources/com/zerocracy/bundles/resigns_stale_users_from_project/_after.groovy[30] UnusedVariable: The variable [farm] in class com.zerocracy.bundles.resigns_stale_users_from_project.None is not used```

**Comment by carlosmiranda** [comment]
@g4s8 this should be fixed now, please check

**Comment by g4s8** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job `gh:zerocracy/farm#1685` is not assigned, can't get performer

**Comment by 0crat** [comment]
The job [#1685](https://github.com/zerocracy/farm/issues/1685) is now out of scope

**Pull request merged**